modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -1002,18 +1007,18 @@ else if (!deleteRelationshipObjectField) {
 			objectFieldLocalService.getObjectFieldsCount(
 				objectField.getObjectDefinitionId(), false);
 
-		if (objectDefinition.isApproved() &&
-			!objectDefinition.isUnmodifiableSystemObject() &&
+		if (objectDefinition.isApproved() && !objectDefinition.isSystem() &&
 			(customObjectFieldsCount == 1)) {
 
 			throw new RequiredObjectFieldException();
 		}
 
-		if (FeatureFlagManagerUtil.isEnabled("LPS-190890") &&
-			objectDefinition.isApproved() && objectDefinition.isModifiable() &&
-			objectDefinition.isSystem()) {
+		if (objectDefinition.isModifiable() && objectField.isSystem() &&
+			!SystemObjectDefinitionsUtil.
+				isAllowedManageSystemObjectDefinitions()) {
 
-			throw new UnsupportedOperationException();
+			throw new ObjectFieldSystemException(
+				"Only allowed bundles can delete system object fields");
```

Comment:
Note: discuss if we need to add a condition when deleting an Object Field of an Unmodifiable Object Definition.
Suggestion: make this condition verify `objectDefinition.isSystem()` instead of `objectDefinition.isModifiable()`

Reply:
makes sense